src/Core/Engine/Systems/LightManager.hpp

```
@@ -3,26 +3,39 @@
     SSBO pointLightSSBO;
     SSBO dirLightSSBO;
     SSBO spotLightSSBO;
-    bool dirty_SSBO = false;
+
+    bool dirty_Point = false;
+    bool dirty_Direction = false;
+    bool dirty_Spot = false;
+
+    const entt::registry& m_SceneRegistry;
+
+    static DirectionLightGPU ConvertLightToData(const DirectionalLight& light);
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)